cid/drainer.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import print_function
+import boto3
+import json
+import logging
+import os
+
+
+class ContainerInstanceDrainer:
+
+    def __init__(self, event, context):
+
+        logging.basicConfig()
+        logger = logging.getLogger()
+        logger.setLevel(getattr(logging, os.environ.get('LOGLEVEL', 'WARNING').upper()))
+        self.logger = logger
+
+        self.event = event
+        self.message = json.loads(event['Records'][0]['Sns']['Message'])
+        self.cache = self.message['_CidLambdaCache'] if '_CidLambdaCache' in self.message else {}
```

Comment:
is it a cache across invocations or within one invocation?

Reply:
The cache is propagated by appending it to the message during the first invocation.